src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs

```diff
@@ -84,6 +82,62 @@ public int TryGetLast(out bool found)
                 found = true;
                 return _end - 1;
             }
+
+            // Destination *must* be non-empty and exactly match the range length
+            private void InitializeSpan(Span<int> destination)
+            {
+                if (destination.Length < Vector<int>.Count * 2)
+                {
+                    int end = _end;
+                    ref int pos = ref MemoryMarshal.GetReference(destination);
+                    for (int cur = _start; cur < end; cur++)
+                    {
+                        pos = cur;
+                        pos = ref Unsafe.Add(ref pos, 1);
+                    }
```

Comment:
You shouldn't need unsafe code here to avoid bounds checks.
```C#
int cur = _start;
for (int i = 0; i < destination.Length; i++)
{
    destination[i] = cur;
    cur++;
}
```